### PR TITLE
Ardupilot mission handling and vehicle type checking

### PR DESCRIPTION
**Pull request by DaRavenox**
Adds support to mission_raw to upload, start missions and clear missions when the autopilot is ArduPilot.

Further adds support to grab the vehicle type from the HEARTBEAT message, and to use this to determine what the correct mode command is for a ArduPilot vehicle, since this depends on the type.

### CODE REVIEW

File: src/core/system_impl.h [hunk 1]
```diff
@@ -50,6 +50,52 @@ class SystemImpl : public Sender {
...
+        Guided_No_GPS = 20,
+        Smart_RTL = 21,
+        Flow_Hold = 22,
+        Follow = 23,
+        Zigzag = 24,
+        System_ID = 25,
+        Auto_Rotate = 26,
+        Auto_RTL = 27,
+        Turtle = 28
+    };
```
Comment (julianoes) [inline src/core/system_impl.h:97]: What about moving these enums to a separate header file like https://github.com/mavlink/MAVSDK/blob/main/src/core/px4_custom_mode.h?

File: src/plugins/mission_raw/mission_raw_impl.cpp [hunk 2]
```diff
@@ -377,15 +399,23 @@ void MissionRawImpl::clear_mission_async(const MissionRaw::ResultCallback& callb
...
     reset_mission_progress();
 
-    _parent->mission_transfer().clear_items_async(
-        MAV_MISSION_TYPE_MISSION, [this, callback](MAVLinkMissionTransfer::Result result) {
-            auto converted_result = convert_result(result);
-            _parent->call_user_callback([callback, converted_result]() {
-                if (callback) {
-                    callback(converted_result);
-                }
+    // For ArduPilot to clear a mission we need to upload an empty mission.
```
Comment (julianoes) [inline src/plugins/mission_raw/mission_raw_impl.cpp:402]: Ha, really? :joy: That's funny.

File: src/core/system_impl.h [hunk 3]
```diff
@@ -50,6 +50,52 @@ class SystemImpl : public Sender {
         Stabilized,
     };
 
+    // Enumeration representing the available modes for the Arudpilot rover autopilot.
+    enum class APRoverMode {
```
Comment (julianoes) [inline src/core/system_impl.h:54]: Usually we try the convention to lowercase acronyms like so: `ApRoverMode`, however, I assume you copied them from somewhere, so then maybe it's better leave em as is.

**Comment by DaRavenox** [comment]
Heya, sorry about the number of changes. Had a chat with Seb and we decided on hiding the ardupilot mode behind flight_mode, so now I translate back and forth from it.

**Review by JonasVautherin — APPROVED** [review]

**Review by julianoes — APPROVED** [review]
Looks good to me. 

Any objections @JonasVautherin?

**Pull request merged**